Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -443,13 +464,40 @@ - (void) receivedProxyHTTPResponse: (CFHTTPMessageRef)httpResponse {
 - (void) receivedHTTPResponse: (CFHTTPMessageRef)httpResponse {
     // Post the response headers to LiteCore:
     NSDictionary *headers =  CFBridgingRelease(CFHTTPMessageCopyAllHeaderFields(httpResponse));
+    
     NSString* cookie = headers[@"Set-Cookie"];
-    if ([cookie containsString: @", "]) {
-        // CFHTTPMessage incorrectly merges multiple Set-Cookie headers. Undo that:
-        NSMutableDictionary* newHeaders = [headers mutableCopy];
-        newHeaders[@"Set-Cookie"] = [cookie componentsSeparatedByString: @", "];
-        headers = newHeaders;
+    if (cookie.length > 0) {
+        // separated by comma will fail, hence regex: https://issues.couchbase.com/browse/CBL-681
+        NSError* error = NULL;
+        NSRegularExpression* r = [NSRegularExpression regularExpressionWithPattern: @"\\,\\s*\\w+="
+                                                                           options: NSRegularExpressionCaseInsensitive
+                                                                             error: &error];
+        NSArray* matches = [r matchesInString: cookie options:0 range: NSMakeRange(0, [cookie length])];
+        NSMutableArray* cookies = [NSMutableArray arrayWithCapacity: matches.count];
+        for (NSTextCheckingResult* match in matches) {
+            if ([match range].length > 0) {
+                NSString* cookieStr = [cookie substringWithRange: [match range]];
+                [cookies addObject: cookieStr];
+                
+                // save to lite core
+                CBL_LOCK(_db) {
+                    C4Error err = {};
+                    CBLStringBytes cookieSlice(cookieStr);
+                    CBLStringBytes host(_remoteURL.host);
+                    CBLStringBytes path(_remoteURL.path.stringByDeletingLastPathComponent);
+                    if (!c4db_setCookie(_db.c4db, cookieSlice, host, path, &err)) {
```

Comment:
Can you check if this requires lock?

Reply:
both set and get are added to db lock. since both has db-access